Sources/BraintreePayPal/BTPayPalAccountNonce.swift

```diff
@@ -0,0 +1,56 @@
+import Foundation
+
+#if canImport(BraintreeCore)
+import BraintreeCore
+#endif
+
+/// Contains information about a PayPal payment method
+@objcMembers public class BTPayPalAccountNonce: BTPaymentMethodNonce {
+    
+    /// Payer's email address.
+    public let email: String?
+    
+    /// Payer's first name.
+    public let firstName: String?
+    
+    /// Payer's last name.
+    public let lastName: String?
+    
+    /// Payer's phone number.
+    public let phone: String?
+    
+    /// The billing address.
+    public let billingAddress: BTPostalAddress?
+    
+    /// The shipping address.
+    public let shippingAddress: BTPostalAddress?
+    
+    /// Client metadata id associated with this transaction.
+    public let clientMetadataID: String?
+    
+    /// Optional. Payer id associated with this transaction.
+    /// Will be provided for Vault and Checkout.
+    public let payerID: String?
+    
+    /// Optional. Credit financing details if the customer pays with PayPal Credit.
+    /// Will be provided for Vault and Checkout.
+    public let creditFinancing: BTPayPalCreditFinancing?
+    
+    init?(json: BTJSON) {
+        guard let nonce = json["nonce"].asString() else { return nil }
+        
+        let details = json["details"]
+        let payerInfo = details["payerInfo"]
+
+        self.email = payerInfo["email"].asString() ?? details["email"].asString()        
```

Comment:
```suggestion
        email = payerInfo["email"].asString() ?? details["email"].asString()        
```
🧶 : take or leave.

Reply:
Seems like we use the explicit `self.` [in Northstar as well](https://github.com/paypal/iOS-SDK/blob/855802216f9fbddbc6709d101e8871b307999628/Sources/PaymentsCore/AnalyticsEventData.swift#L88). I personally prefer `self.` but my tastes can be a little wordy. What do others think?

Reply:
I personally am a fan of removing the explicit self where possible to clean things up, but we have done a mix of both throughout the Swift conversion. For example the [SEPA nonce](https://github.com/braintree/braintree_ios/blob/c449e0b81c78fcae47bb02cfb8f987aed2a19996/Sources/BraintreeSEPADirectDebit/BTSEPADirectDebitNonce.swift#L23-L25) does use explicit self while the [PayPalNativeCheckout nonce](https://github.com/braintree/braintree_ios/blob/c449e0b81c78fcae47bb02cfb8f987aed2a19996/Sources/BraintreePayPalNativeCheckout/BTPayPalNativeCheckoutAccountNonce.swift#L47-L52) does not. This is certainly something that we can solidify and implement in our style guide either was we decide.

Reply:
It seems like generally we avoid using `self` when possible in most methods, but we tend to use it in initializers. I personally like the explicitness when initializer parameter names are the same as property names (e.g. `self.firstName = firstName`) but am open to whatever.

Reply:
I am down with this as well as our style! Do we have a plan to "officially" move our style guide into the repo at some point?

Reply:
Agreed yeh, we should move our style guide out of HackMD soon! Maybe the knowledge repo, or maybe the real repo.